time_manager/get_cal_time.F90

```diff
@@ -300,47 +309,63 @@ function get_cal_time(time_increment, units, calendar, permit_calendar_conversio
   base_time             = set_date(year+floor(time_increment)  ,month,day,hour,minute,second)
   base_time_plus_one_yr = set_date(year+floor(time_increment)+1,month,day,hour,minute,second)
   call get_time(base_time_plus_one_yr - base_time, second, day)
-  dt = (day*86400+second)*(time_increment-floor(time_increment))
-  increment_days = floor(dt/86400)
-  increment_seconds = int(dt - increment_days*86400)
+  dt = real(day*86400+second, r8_kind)*(time_increment-real(floor(time_increment), r8_kind))
+  increment_days = floor(dt/86400.0_r8_kind)
+  increment_seconds = int(dt - real(increment_days*86400, r8_kind))
 else if(lowercase(units(1:12)) == 'months since') then
-  month_fraction = time_increment - floor(time_increment)
+  month_fraction = time_increment - real(floor(time_increment), r8_kind)
   increment_years  = floor(time_increment/12)
   increment_months = floor(time_increment) - 12*increment_years
   call get_date(base_time, year,month,day,hour,minute,second)
   base_time = set_date(year+increment_years,month+increment_months  ,day,hour,minute,second)
-  dt = 86400*days_in_month(base_time) * month_fraction
+  dt = real( 86400*days_in_month(base_time), r8_kind) * month_fraction
   increment_days = floor(dt/86400)
```

Comment:
one more 86400._r8_kind

Reply:
that line should be fixed, maybe the comparison glitched out:

https://github.com/rem1776/FMS/blob/aa84fd72e47318d397776bd42f0d2a1bd497a551/time_manager/get_cal_time.F90#L324